javatools/src/main/java/org/xvm/compiler/ast/AssignmentStatement.java

```diff
@@ -650,9 +648,8 @@ else if (fConditional)
                     atypeReq[0] = pool().typeBoolean();
                     System.arraycopy(atypeLVals, 0, atypeReq, 1, cLVals);
 
-                    exprRightNew = exprRight.validateMulti(ctxRValue, atypeReq, errs);
-
-                    if (exprRightNew == null)
+                    exprRight = rvalue.validateMulti(ctxRValue, atypeReq, errs);
+                    if (exprRight == null)
                         {
                         return null;
```

Comment:
I'm not sure why we "return null" from here rather than "break". Returning here skips the "nodeLeft.resetLValueTypes(ctx);" call which happens on all other error paths

Reply:
True, but `TypeConstant[] atypeAll = exprRight.getTypes();` would be an automatic NPE ...